internal/resources/ekscluster/data_source.go

```diff
@@ -79,6 +82,36 @@ func dataSourceTMCEKSClusterRead(ctx context.Context, d *schema.ResourceData, m
 			return true, nil
 		}
 
+		if isWaitForKubeconfig(d) {
+			clusFullName := &clustermodel.VmwareTanzuManageV1alpha1ClusterFullName{Name: resp.EksCluster.Spec.AgentName, OrgID: clusterFn.OrgID, ManagementClusterName: "eks", ProvisionerName: "eks"}
+			clusterResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(clusFullName)
+			// nolint: wsl
+			if err != nil {
+				log.Printf("Unable to get Tanzu Mission Control cluster entry, name : %s", clusterFn.Name)
+				return true, nil
```

Comment:
Return the error too. In case this fails with the timeout, we can see the error. Also, log the error.

Reply:
@asutoshpalai  fixed it